### PR TITLE
More explicit/aggressive 404s

### PR DESCRIPTION
**Pull request by jtart**
Currently, After.JS will send false positives by:
- sending a 200 even if there is no router match (could fall through to a RR catch-all/not found route and still return a 200)
- sending HTML to the client even if React hasn't rendered anything/the main body is empty

This PR:
- sets the response status to 404 if there is no router match
- sends an explicit 404 to the client if no main body HTML is rendered

### CODE REVIEW

File: src/render.tsx [hunk 1]
```diff
@@ -85,6 +90,10 @@ export async function render<T>(options: AfterRenderProps<T>) {
     match: reactRouterMatch,
   });
 
+  if(!html || html.length === 0) {
```
Comment (jaredpalmer) [inline src/render.tsx]: not sure i understand this one.

Comment (jtart) [reply src/render.tsx]: if not html is rendered/returned from the `renderToString`, will return a 404.

Comment (jtart) [reply src/render.tsx]: Further to this, if we do not send this explicit 404 to the client, anything inside Document will be sent to the client.

Comment (stepanh) [reply src/render.tsx]: Hi @jtart - does this solve your use case?

Right now, you can define a catch-all path in your `routes` like:
```
  // add this as the last route
  {
    path: '/', // you need to define path for getInitialProps to work in component
    component: NotFound,
    exact: false,
  },
```
and the NotFound component can look like this:
```
class NotFound extends React.Component {
  static async getInitialProps(ctx: any) {
    if (ctx.res) {
      ctx.res.statusCode = 404;
    }
  }

  render() {
    return (
      <div>
        <Header />
        <h1>404 Not Found</h1>
        <p>Page not found</p>
        <Footer />
      </div>
    );
  }
}
```
IMHO nothing more is needed.

**Pull request closed without merging**